iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/Properties.java

```diff
@@ -0,0 +1,171 @@
+package com.powsybl.iidm.network.util;
+
+import com.google.common.collect.Sets;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.DanglingLine;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class Properties {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Properties.class);
+
+    private static final String INCONSISTENCY_WARN_EMPTY_SIDE_1 = "Inconsistencies of property '{}' between both sides of merged line. Side 1 is empty, keeping side 2 value '{}'";
+    private static final String INCONSISTENCY_WARN_EMPTY_SIDE_2 = "Inconsistencies of property '{}' between both sides of merged line. Side 2 is empty, keeping side 1 value '{}'";
+    private static final String INCONSISTENCY_ERROR_BOTH_SIDES = "Inconsistencies of property '{}' between both sides of merged line. '{}' on side 1 and '{}' on side 2. Removing the property of merged line";
+
+    public static class Property {
+        private final Pair<Type, Object> entry;
+
+        public Property(Type type, Object value) {
+            entry = new ImmutablePair<>(type, value);
+        }
+
+        public Property(Object value) {
+            Type type;
+            if (value instanceof String) {
+                type = Type.STRING;
+            } else if (value instanceof Boolean) {
+                type = Type.BOOLEAN;
+            } else if (value instanceof Integer) {
+                type = Type.INTEGER;
+            } else if (value instanceof Double) {
+                type = Type.DOUBLE;
+            } else {
+                throw new PowsyblException("Invalid property type, only available types are : STRING, INTEGER, DOUBLE, BOOLEAN");
+            }
+            entry = new ImmutablePair<>(type, value);
+        }
+
+        public Type getType() {
+            return entry.getKey();
+        }
+
+        public Object getValue() {
+            return entry.getValue();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Property property = (Property) o;
+            return Objects.equals(entry, property.entry);
```

Comment:
`return Objects.equals(entry, property.entry)` seems to be sufficient: the entry contains the type and the value, and `o` knows its own class